.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build
+on: [push]
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      # Cache
+      - name: Cache .gradle/caches
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-cache-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle-cache-
+      - name: Cache .gradle/wrapper
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle-wrapper-
+      # Licensing
+      - name: Licensing
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: --console=plain license
+      # Coding style
+      - name: Coding style
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: --console=plain codenarcAll
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v1
+      # use a short path to avoid exceeding the path limit on Windows... sigh
+      with:
+        path: 'w'
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    # Cache
+    - name: Cache .gradle/caches
+      uses: actions/cache@v1
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-cache-${{ hashFiles('**/*.gradle') }}
+        restore-keys: ${{ runner.os }}-gradle-cache-
+    - name: Cache .gradle/wrapper
+      uses: actions/cache@v1
+      with:
+        path: ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('**/*.gradle') }}
+        restore-keys: ${{ runner.os }}-gradle-wrapper-
+    - name: Build and test
+      uses: eskatos/gradle-command-action@v1
+      with:
+        arguments: --console=plain --scan test intTest remoteTest gradleTest jacocoRootReport
```

Comment:
We might break down this step into multiple tasks:

- unit test
- integration test
- remote test
- gradle test

Not sure if I need to add `jacocoRootReport` each time?
Speaking of `jacocoRootReport` we should probably upload the artifact otherwise it's not really useful? Or perhap the jacoco report is available on https://scans.gradle.com ?

Reply:
Look at the .gitlab-yml file. It has an extensive workflow (except that it is all on Linux). 

Also we do not need to run everything on every platform. So basically if we can use the workflow that exists on Appveyor on Mac & Windows and run the full workflow on Linux it should provide adequate coverage.

> Not sure if I need to add jacocoRootReport each time

Basically the reports should only be generated at the end, when all of the tests and integration tests have finished.